lua/astrocommunity/note-taking/venn-nvim/init.lua

```diff
@@ -0,0 +1,39 @@
+return {
+  "jbyuki/venn.nvim",
+  event = "VeryLazy",
+  cmd = "VBox",
+  keys = {
+    {
+      "<Leader>vn",
+      "<Cmd>lua Toggle_venn()<cr>",
+      { silent = true },
+      desc = "Toggle Venn diagram",
+    },
+  },
+  config = function()
+    function _G.Toggle_venn()
+      local venn_enabled = vim.inspect(vim.b.venn_enabled)
+      if venn_enabled == "nil" then
+        vim.notify("enabled Venn mode", "info", { title = "Venn" })
+        vim.b.venn_enabled = true
+        vim.cmd [[setlocal ve=all]]
+        -- draw a line on HJKL keystokes
+        vim.api.nvim_buf_set_keymap(0, "n", "J", "<C-v>j:VBox<CR>", { noremap = true })
+        vim.api.nvim_buf_set_keymap(0, "n", "K", "<C-v>k:VBox<CR>", { noremap = true })
+        vim.api.nvim_buf_set_keymap(0, "n", "L", "<C-v>l:VBox<CR>", { noremap = true })
+        vim.api.nvim_buf_set_keymap(0, "n", "H", "<C-v>h:VBox<CR>", { noremap = true })
+        -- draw a box by pressing "f" with visual selection
+        vim.api.nvim_buf_set_keymap(0, "v", "f", ":VBox<CR>", { noremap = true })
+      else
+        vim.notify("disabled Venn mode", "info", { title = "Venn" })
+        vim.cmd [[setlocal ve=]]
+        vim.api.nvim_buf_del_keymap(0, "n", "J")
+        vim.api.nvim_buf_del_keymap(0, "n", "K")
+        vim.api.nvim_buf_del_keymap(0, "n", "L")
+        vim.api.nvim_buf_del_keymap(0, "n", "H")
+        vim.api.nvim_buf_del_keymap(0, "v", "f")
+        vim.b.venn_enabled = nil
+      end
+    end
+  end,
```

Comment:
Please set the mappings with astrocore.

Reply:
I've pushed the changes to set the mappings. However there is no astrocore API to remove the mappings. So leaving the remove part as is.